code/controllers/subsystems/cargo.dm

```diff
@@ -656,6 +664,19 @@ var/datum/controller/subsystem/cargo/SScargo
 				continue
 			clear_turfs += T
 
+	for(var/E in queued_mining_equipment)
+		if(!ispath(E))
+			continue
+
+		var/i = rand(1,clear_turfs.len)
+		var/turf/pickedloc = clear_turfs[i]
+		clear_turfs.Cut(i,i+1)
+
+		if(isturf(pickedloc))
+			new E(pickedloc)
+
+	queued_mining_equipment.Cut()
```

Comment:
It might be better to remove the specific equipment entry after each successful spawn instead of clearing the whole list at the end.